README.md

```diff
@@ -39,23 +45,29 @@ git clone https://github.com/obsproject/obs-deps.git
 ```
 
 * Enter the `obs-deps` directory
-* (*Optional*) Create the build scripts by running `./build_script_generator.py .github/workflows/build_deps.yml`
-* Run `bash ./build-script-macos-01.sh` to build main dependencies
-* Run `bash ./build-script-macos-02.sh` to build Qt dependency
+* Run `bash ./CI/build-deps-macos.sh` to build main dependencies
+* Run `bash ./CI/build-qt-macos.sh` to build Qt dependency
 
 ### Usage
 
-* Common directory to place obs-deps is `/tmp/obsdeps/`, but a custom path can be used
-* Unpack pre-built dependencies to `/tmp` by running `tar -xf ./macos-deps-VERSION.tar.gz -C /tmp` (replace `VERSION` with the downloaded/desired version)
-* Unpack pre-built Qt dependency to `/tmp` by running `tar -xf ./macos-qt-QT_VERSION-VERSION.tar.gz -C /tmp` (replace `QT_VERSION` and `VERSION` with the downloaded/desired versions)
-* **IMPORTANT:** Remove the quarantine attribute from the downloaded Qt dependencies by running `xattr -r -d com.apple.quarantine /tmp/obsdeps`
-* Use `/tmp/obsdeps` as `QTDIR`, `SWIGDIR`, and `DepsPath` when invoking `cmake` for OBS:
+* Create a destination directory for the dependencies (e.g. `obs-deps`)
+* Unpack the dependencies into this directory (e.g. via `XZ_OPT=-T0 tar -xf macos-deps-VERSION-universal.tar.xz -C obs-deps` - replace `VERSION` with the downloaded/desired version)
+* Repeat the same for the Qt dependencies
+* **IMPORTANT:** Remove the quarantine attribute from the downloaded Qt dependencies by running `xattr -r -d com.apple.quarantine obs-deps`
+* Use `obs-deps` as part of `CMAKE_PREFIX_PATH` when running `cmake` for OBS:
 
 ```
-cmake -DCMAKE_OSX_DEPLOYMENT_TARGET=10.13 -DQTDIR="/tmp/obsdeps" -DSWIGDIR="/tmp/obsdeps" -DDepsPath="/tmp/obsdeps" [..]
+cmake -DCMAKE_PREFIX_PATH="some_other_path;obs-deps" [..]
 ```
 
 ## Contributing
 
-* Do not make changes to the `build-script-{macos/ubuntu/windows}-{num}.sh` scripts, which are generated by `.github/workflows/create_scripts.yml`
-* Instead, edit `.github/workflows/build_deps.yml`
\ No newline at end of file
+* Add/edit seperate build scripts for every dependency in the `CI/[OPERATING SYSTEM/` directory
+* For new dependencies:
+    * Create the `sha256sum` of the downloaded dependency archive
+    * Add the dependency version as `[DEPENDENCY_NAME]_VERSION` and the downloaded archive hash as `[DEPENDENCY_NAME]_HASH` to the Github actions workflow in `.github/workflows/main.yml` as well as the main build script in `CI/build-deps-macos.sh`.
+* For existing dependencies:
+    * Always update the `sha256sum` of the updated dependency archive as well as the version
+* Patches need to be placed either in the patches directory (if applied for all OS') or inside the patches directory for a specific OS
```

Comment:
`OS'` doesn't need an apostrophe by itself.  If it were `OS's`, then maybe, but not required, according to APA guidelines.  I'll leave the choice between `OS` vs. `OSs` vs. `OS's` up to you.